lib/Basics/process-utils.cpp

```diff
@@ -960,6 +961,31 @@ void TRI_CreateExternalProcess(char const* executable,
   }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief Reads from the pipe of processes
+////////////////////////////////////////////////////////////////////////////////
+
+bool TRI_ReadPipe(ExternalProcess const* process,
+                  char* buffer,
+                  size_t &bufferSize) {
+  if (process == nullptr || TRI_IS_INVALID_PIPE(process->_readPipe)) {
+    return false;
+  }
+
+  memset(buffer, 0, bufferSize);
+
+#ifndef _WIN32
+  bool haveMore = TRI_ReadPointer(process->_readPipe, buffer, bufferSize);
+#else
+  bool haveMore = TRI_READ_POINTER(process->_readPipe, buffer, bufferSize);
+#endif
+  if (!haveMore) {
+    bufferSize = strlen(buffer);
```

Comment:
This is still unsafe. The buffer is zero-filled initially, but what happens if `bufferSize` bytes are read, the stream ends and the last byte is not `0`? Then the `strlen` call in line 983 is not safe.

Reply:
modified the read pointer function to set the length it actually read.